packages/flutter_tools/test/general.shard/channel_test.dart

```diff
@@ -60,13 +60,28 @@ void main() {
       await simpleChannelTest(<String>['channel', '-v']);
     });
 
+    testUsingContext('list sorted by stability', () async {
+      await simpleChannelTest(<String>['channel']);
+
+      final List<String> tmpChannels = testLogger.statusText.toString().split('\n');
+      final List<String> channels = tmpChannels.sublist(1, tmpChannels.length - 1);
+
+      expect(channels.length, 4);
+      expect(channels[0], contains('master'));
+      expect(channels[1], contains('dev'));
+      expect(channels[2], contains('beta'));
+      expect(channels[3], contains('stable'));
+    });
+
     testUsingContext('removes duplicates', () async {
       final Process process = createMockProcess(
           stdout: 'origin/dev\n'
                   'origin/beta\n'
+                  'origin/master\n'
```

Comment:
What happens here if master is not included?

Reply:
It will not print nothing, since the print happens only after that the process found all the official channels (and sort them).

```dart
if (!done && officialChannels.length == FlutterVersion.officialChannels.length) {
   //...order then print...
}
```

Which is not ideal, but not bad either IMO.

At the end `flutter channel` is expected to print the list of official channels so if `master` is not in the output of `git branch -r` but still in the `FlutterVersion.officialChannels` i think that this will raise up an underlying issue with the flutter repository. 

Sure is that maybe instead of simply silently fails we can improve with en error message in this case.

Reply:
This change needs to be resilient to missing channels, it sounds like it will fail if run on a fork of flutter that is missing the exact branches.

Reply:
Actually i don't think it will cause any issue because even in the case of a forked repo without the official channels `git branch -r` will found the `upstream` ones anyway, matching with the official channels.

I tried this myself and it not cause any issue, this remove duplicates function is actually addressing exactly this scenario in which both the `upstream` and the `origin` have the same branch names.

Reply:
@jonahwilliams Any thought on this? I'll be more the fine to implement your proposed change, but i'll need a bit of guidance then.